packages/cryptography/src/primitive/ecdsa.js

```diff
@@ -12,20 +11,43 @@ const secp256k1 = new elliptic.ec("secp256k1");
  * @returns {KeyPair}
  */
 export function generate() {
-    const keypair = secp256k1.genKeyPair();
+    for (let attempt = 0; attempt < 30; attempt++) {
+        const privateKey = random.bytes(32);
+        const publicKey = secp256k1.pointFromScalar(privateKey, true);
 
-    return {
-        privateKey: hex.decode(keypair.getPrivate("hex")),
-        publicKey: hex.decode(keypair.getPublic(true, "hex")),
-    };
+        if (!secp256k1.isPrivate(privateKey) || publicKey == null) {
+            continue;
+        }
+
+        return {
+            privateKey,
+            publicKey,
+        };
+    }
+
+    throw new Error(`Couldn't generate valid data.`);
 }
 
 /**
  * @returns {Promise<KeyPair>}
  */
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 export async function generateAsync() {
-    return Promise.resolve(generate());
+    for (let attempt = 0; attempt < 30; attempt++) {
+        const privateKey = await random.bytesAsync(32);
+        const publicKey = secp256k1.pointFromScalar(privateKey, true);
+
+        if (!secp256k1.isPrivate(privateKey) || publicKey == null) {
+            continue;
+        }
+
+        return {
+            privateKey,
+            publicKey,
+        };
+    }
+
+    throw new Error(`Couldn't generate valid data.`);
```

Comment:
nit:
```suggestion
    throw new Error(`Couldn't generate valid ecdsa key pair.`);
```

Reply:
No longer applicable after I've migrated back to `elliptic` instead of `tiny-secp256k1`